services/motion/builtin/move_attempt.go

```diff
@@ -49,6 +50,18 @@ func newMoveAttempt(ctx context.Context, request *moveRequest) *moveAttempt {
 	var waypointIndex atomic.Int32
 	waypointIndex.Store(1)
 
+	// effectively don't poll if the PositionPollingFreqHz is not provided
+	positionPollingFreq := time.Duration(math.MaxInt64)
+	if request.config.PositionPollingFreqHz > 0 {
+		positionPollingFreq = time.Duration(1000/request.config.PositionPollingFreqHz) * time.Millisecond
+	}
+
+	// effectively don't poll if the ObstaclePollingFreqHz is not provided
+	obstaclePollingFreq := time.Duration(math.MaxInt64)
+	if request.config.ObstaclePollingFreqHz > 0 {
+		obstaclePollingFreq = time.Duration(1000/request.config.ObstaclePollingFreqHz) * time.Millisecond
+	}
+
```

Comment:
This will make it so that the obstacle & position pollers won't poll if their polling frequencies are 0.